### PR TITLE
Add pattern matching to field values

### PR DESCRIPTION
**Pull request by mtojek**
Issue: https://github.com/elastic/elastic-package/issues/113

This PR adds pattern matching to field values. A pattern can specified in the field definition in the `fields.yml` file.

### CODE REVIEW

**Comment by mtojek** [comment]
Draft because of waiting for this one first: https://github.com/elastic/package-spec/pull/80

File: internal/fields/validate.go [hunk 1]
```diff
@@ -190,7 +190,19 @@ func parseElementValue(key string, definition fieldDefinition, val interface{})
...
+		if !valid || definition.Pattern == "" {
+			break
+		}
+
+		valid, err := regexp.MatchString(definition.Pattern, valStr)
+		if err != nil {
+			return errors.Wrap(err, "invalid pattern")
+		}
+		if !valid {
+			return fmt.Errorf("field \"%s\" has invalid format, pattern: %s, actual: %s", key, definition.Pattern, valStr)
```
Comment (ycombinator) [inline internal/fields/validate.go:204]: I think we need to start making our error messages, especially validation ones, more user-friendly (related: https://github.com/elastic/package-spec/issues/75). So I would suggest changing this to something like:

```suggestion
			return fmt.Errorf("field \"%s\"''s value, %s, does not match the expected pattern: %s", key, valStr, definition.Pattern)
```

Comment (mtojek) [reply internal/fields/validate.go]: Good idea, I updated the PR.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**